qetpy/core/didv/_didv.py

```diff
@@ -911,107 +948,63 @@ def calc_smallsignal_params(self, ivsweep_results,
         self._rp = biasparams_dict['rp']
 
 
-        # calculate small signal parameters with proper bias parameters
-        if calc_2poles:
-            self._calc_ssp(2,
-                           biasparams_dict=biasparams_dict.copy(),
-                           lgc_ssp_light=False)
+        # calculate small signal parameters
 
-        if calc_3poles:
-            self._calc_ssp(3,
+        # loop poles
+        for model_poles in model_list:
+
+            # calc
+            self._calc_ssp(model_poles,
                            biasparams_dict=biasparams_dict.copy(),
                            lgc_ssp_light=True)
-
-
-        # Check if infinite loop gain needs to be done
-        # if inf_loop_gain_approx == 'auto' AND lopp gain negative
-        #     -> automatically make approximation that loop gain infinite 
             
+            # Check if infinite loop gain needs to be done
+            # if inf_loop_gain_approx == 'auto' AND lopp gain negative
+            #     -> automatically make approximation that loop gain infinite 
 
-        # 2-poles
-        if calc_2poles:
-
+            results = copy.deepcopy(self._fit_results[model_poles])
+                            
             # check if infinite loop gain needs to be done
             set_infinite_loop_gain = inf_loop_gain_approx
             if inf_loop_gain_approx == 'auto':
```

Comment:
Where is this being set? I don't see it as a variable you can set when calling dofit

Reply:
You mean the variable "model_poles"? It is in a loop:
for model_poles in model_list:
      ...


model_list  can be [2,3 ] or just [2] or [3] depending is dofit(2) and/or dofit(3) wa called before.  Given that the same functions are now used for 2 and 3, we don't need to have anything specific for 2/3 as before so we can loop a loop